# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import math
from math import sqrt
from scipy.stats import t

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
data.describe()

education       ofjobs     yearsexp       honors    volunteer  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      3.618480     3.661396     7.842916     0.052772     0.411499   
std       0.714997     1.219126     5.044612     0.223601     0.492156   
min       0.000000     1.000000     1.000000     0.000000     0.000000   
25%       3.000000     3.000000     5.000000     0.000000     0.000000   
50%       4.000000     4.000000     6.000000     0.000000     0.000000   
75%       4.000000     4.000000     9.000000     0.000000     1.000000   
max       4.000000     7.000000    44.000000     1.000000     1.000000   

          military     empholes  occupspecific   occupbroad  workinschool  \
count  4870.000000  4870.000000    4870.000000  4870.000000   4870.000000   
mean      0.097125     0.448049     215.637782     3.481520      0.559548   
std       0.296159     0.497345     148.127551     2.038036      0.496492   
min       0.000000     0.000000       7.000000     1.000000      0.000000   
25%       0.000000     0.000000      27.000000     1.000000      0.000000   
50%       0.000000     0.000000     267.000000     4.000000      1.000000   
75%       0.000000     1.000000     313.000000     6.000000      1.000000   
max       1.000000     1.000000     903.000000     6.000000      1.000000   

          ...           educreq      compreq       orgreq        manuf  \
count     ...       4870.000000  4870.000000  4870.000000  4870.000000   
mean      ...          0.106776     0.437166     0.072690     0.082957   
std       ...          0.308866     0.496083     0.259649     0.275854   
min       ...          0.000000     0.000000     0.000000     0.000000   
25%       ...          0.000000     0.000000     0.000000     0.000000   
50%       ...          0.000000     0.000000     0.000000     0.000000   
75%       ...          0.000000     1.000000     0.000000     0.000000   
max       ...          1.000000     1.000000     1.000000     1.000000   

          transcom     bankreal        trade   busservice   othservice  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      0.030390     0.085010     0.213963     0.267762     0.154825   
std       0.171677     0.278932     0.410141     0.442847     0.361773   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           missind  
count  4870.000000  
mean      0.165092  
std       0.371308  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 55 columns]

Let's confine our analysis to race factor only

In [6]:
df = data[['race','call']]
df.head()

race  call
0    w   0.0
1    w   0.0
2    b   0.0
3    b   0.0
4    w   0.0

We will check proportion of calls in the white(w) and black(b) populations

In [7]:
# Compute proportion of call backs for each group
# the w and b datasets
df_w = df[df['race'] == 'w']
df_b = df[df['race'] == 'b']
# Sample size for each type
n_w = len(df_w)
n_b = len(df_b)
print("white count " + str(n_w))
print("black count " + str(n_b))

# number that received call
n_call_w = np.sum(df_w.call)
n_call_b = np.sum(df_b.call)
# Proportion
p_w = n_call_w/n_w
p_b = n_call_b/n_b
[p_w, p_b]

white count 2435
black count 2435


[0.096509240246406572, 0.064476386036960986]

Approximately 9.7% of white resumes got call back compared to 6.4% for black sounding resumes
Now, the question is : Is this difference statictically significant?
The Null Hypothesis: There is no difference between the black and white sounding resumes
The alternate hypothesis : There is significant difference
We will do a two sample t test to determine if the two samples are different. We will use 95% confidence interval


In [8]:
stats.ttest_ind(df_w.call,df_b.call,equal_var=False)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9429415136459352e-05)

The two sample returns a t value of 4.11 and a p value which is very small. Hence the null hypothesis is rejected.
We conclude that there is a signifcant difference between the means.

 -1.What test is appropriate for this problem? Does CLT apply?

A two sample T test is appropriate for this problem. Yes, CLT applies

 -2. What are the null and alternate hypotheses?

The Null Hypothesis: There is no difference between the black and white sounding resumes 
    
The alternate hypothesis : There is significant difference 

 -3. Compute margin of error, confidence interval, and p-value.

At 95% confidence interval the p-value was very small

In [9]:
w_mean = df_w.call.mean()
b_mean = df_b.call.mean()
diff_mean = w_mean - b_mean
print("Diff in means " + str(diff_mean))

Diff in means 0.03203285485506058


In [10]:
# Calculate the margin of error
N1 = 2435
N2 = 2435
degf = (N1 + N2 - 2)
std1 = df_w.call.std()
std2 = df_b.call.std()
std_N1N2 = sqrt( ((N1 - 1)*(std1)**2 + (N2 - 1)*(std2)**2) / degf) 

diff_mean = w_mean - b_mean
MoE = t.ppf(0.975, degf) * std_N1N2 * sqrt(1/N1 + 1/N2)
print("MOE " + str(MoE))


MOE 0.01526193185


At alpha = 0.05, and margin of error is 0.015, Confidence interval is Expected Diff in means +_ margin of error, which is:
0.015, -0.015


 -4. Write a story describing the statistical significance in the context or the original problem.

Racial discrimination is currently an issue in the US job market. In this report we investigated to see whether the data agrees with this. 
A two sample t test between the black and white samples was ran to see if the differences between the callback means are statistically significant.
A very low p value was calculated which means our null hypothesis can be rejected. We can conclude that the race/name of a job applicant does affect the callback rate

 -5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?


The analysis does not necessarily mean that race/name is the most important factor in callback success. 
From the analysis, we can only confirm that race/name is one of the factors that affect callback.

In this analysis I only considered the factor of race; I would have to examine the other variables to determine which is the most immportant factor. For that we may need to consider factors like education, years of experience, honors etc. Each of those factors need to be analyzed for statistical significance.